Added edit functionality

stefanhofman opened the pull request:
This is part of #53, while long-pressing a message and edit option will appear and you will then be able to edit the message.

I am not sure if this is the way to send an edit, but I couldn't find a better way to do it.
```Swift
func edit(text: String, eventId: String) {
        guard !text.isEmpty else { return }
        //swiftlint:disable:next redundant_optional_initialization
        var localEcho: MXEvent? = nil
        let messageObject: [String: Any] = [
            "body": "*" + text,
            "m.new_content": [
                "body": text,
                "msgtype": "m.text"
            ],
            "m.relates_to": [
                "event_id": eventId,
                "rel_type": "m.replace"
            ],
            "msgtype": "m.text"
        ]
        // TODO: Use localEcho to show sent message until it actually comes back
        room.sendMessage(withContent: messageObject, localEcho: &localEcho) { response in
            print(response)
            self.objectWillChange.send()
        }
        self.objectWillChange.send()
    }
```

kiliankoe commented:
You're on a roll! I'll have a look at this in a bit, but it's already looking great 😍

regexident commented:
Great work!

Just to make sure it doesn't fall through the gaps while merging:
The `print(response)` probably should not be merged, as otherwise we would be spilling unencrypted messages into the system logs, that can easily be extracted without authentication.

stefanhofman commented:
> Great work!
> 
> Just to make sure it doesn't fall through the gaps while merging:
> The `print(response)` probably should not be merged, as otherwise we would be spilling unencrypted messages into the system logs, that can easily be extracted without authentication.

I removed the `print(response)` statement in eb529a1, I also found a couple more which I also deleted.

kiliankoe commented on Nio/Conversations/RoomView.swift:
```diff
@@ -56,6 +59,11 @@ struct RoomView: View {
+    var onEdit: (String, String) -> Void
+
+    @State private var edited = false
+    @State private var editEventId = ""
```
Would it make sense to make this of type `String?` instead? So as to better represent that no event is currently being edited? That way we could also remove `edited` entirely, since `editEventId` either holds an event ID meaning we're in edit mode, or it doesn't meaning we're not in edit mode.

stefanhofman replied:
Yes that is better, should be fixed in 6bb8bbf

kiliankoe commented on Nio/Matrix Wrapper/NIORoom.swift:
```diff
@@ -72,12 +71,34 @@ class NIORoom: ObservableObject {
+            "body": "*" + text,
+            "m.new_content": [
+                "body": text,
+                "msgtype": "m.text"
```
I'd suggest using the available constants for the message type here and below.
```suggestion
                "msgtype": kMXMessageTypeText
```

stefanhofman replied:
That is much better indeed, changed it in 50d578f

kiliankoe commented on Nio/Matrix Wrapper/NIORoom.swift:
```diff
@@ -72,12 +71,34 @@ class NIORoom: ObservableObject {
+            ],
+            "m.relates_to": [
+                "event_id": eventId,
+                "rel_type": "m.replace"
```
```suggestion
                "rel_type": MXEventRelationTypeReplace
```

kiliankoe commented on Nio/Matrix Wrapper/NIORoom.swift:
```diff
@@ -72,12 +71,34 @@ class NIORoom: ObservableObject {
+                "event_id": eventId,
+                "rel_type": "m.replace"
+            ],
+            "msgtype": "m.text"
```
```suggestion
            "msgtype": kMXMessageTypeText
```

kiliankoe commented:
It's odd that we have to build the edit message manually, but it's the same for reactions 🤷‍♀️ I've put that into an extra `ReactionEvent` type, if you want you can do the same here, but I think it would make sense to refactor some of that at some point anyway, so feel free to skip that for now.

kiliankoe commented:
Awesome, thank you 🎉
